cmd/eksctl-anywhere/cmd/createcluster.go

```diff
@@ -64,6 +68,21 @@ func init() {
 	createClusterCmd.Flags().StringVar(&cc.managementKubeconfig, "kubeconfig", "", "Management cluster kubeconfig file")
 	createClusterCmd.Flags().StringVar(&cc.installPackages, "install-packages", "", "Location of curated packages configuration files to install to the cluster")
 
+	createClusterCmd.Flags().StringVar(&cc.timeoutOptions.cpWaitTimeout, cpWaitTimeoutFlag, clustermanager.DefaultControlPlaneWaitStr, "Override the default control plane wait timeout (60m).")
+	if err := createClusterCmd.Flags().MarkHidden(cpWaitTimeoutFlag); err != nil {
+		logger.V(5).Info("Warn: Failed to mark flag as hidden: " + cpWaitTimeoutFlag)
+	}
+
+	createClusterCmd.Flags().StringVar(&cc.timeoutOptions.externalEtcdWaitTimeout, externalEtcdWaitTimeoutFlag, clustermanager.DefaultEtcdWaitStr, "Override the default external etcd wait timeout (60m)")
+	if err := createClusterCmd.Flags().MarkHidden(externalEtcdWaitTimeoutFlag); err != nil {
+		logger.V(5).Info("Warn: Failed to mark flag as hidden: " + externalEtcdWaitTimeoutFlag)
+	}
+
+	createClusterCmd.Flags().StringVar(&cc.timeoutOptions.perMachineWaitTimeout, perMachineWaitTimeoutFlag, clustermanager.DefaultPerMachineWaitStr, "Override the default machine wait timeout (10m)/per machine ")
+	if err := createClusterCmd.Flags().MarkHidden(perMachineWaitTimeoutFlag); err != nil {
+		logger.V(5).Info("Warn: Failed to mark flag as hidden: " + perMachineWaitTimeoutFlag)
+	}
```

Comment:
These are duplicated in the upgrade command, could we move them to a common place?

Reply:
Yeah that bothered me as well.. We actually have this problem sprinkled through all the commands with similar args. We duplicate them in each command.  Let me give it some thought on how to use shared options.

Reply:
What about this: a method where you can pass a cobra command and it sets the timeout flags

like `setupTimeoutHiddenFlags(command *cobra.Command)`

It's not the most elegant solution but it solved the duplication and it's quite easy to change/undo in the future.